build-system/tasks/changelog.js

```diff
@@ -306,8 +306,11 @@ function onGitLogSuccess(gitMetadata, logs) {
   assert(typeof logs == 'string', 'git log should be a string.\n' + logs);
   return commits
     .filter(function(commit) {
-      // filter non Pull request merges
-      return commit.indexOf('Merge pull') == 0;
+      // Filter non Pull request merges and also filter
+      // PR's that have merged from some master as it is usually a merge
+      // from our master making it cyclical. (just screws up the PR api requests
+      // to github)
+      return commit.indexOf('Merge pull') == 0 && !(/\/master$/.test(commit));
```

Comment:
actually ive removed this code and changed the git command to:

`git log canary...some-tag --merges --first-parent`

which will prevent those commits from showing up in the commit list i get